tests/test_modules.py

```diff
@@ -6,17 +6,62 @@
 from pggan.models import Generator
 from pggan.models import Discriminator
 from pggan.models.modules import MinibatchStdDev
+from pggan.models.modules import PixelwiseNorm
 
 
 class TestGenerator(unittest.TestCase):
-    def setUp(self):
-        self.generator = Generator()
-        self.sample_input = torch.from_numpy(
-            np.random.randn(512, 1, 1)).float()
-
-    def test_output_shape(self):
-        self.assertEqual(torch.Size([1, 3, 1024, 1024]),
-                         self.generator(torch.unsqueeze(self.sample_input, 0)).shape)
+    def test_get_out_channels(self):
+        resolutions = [1, 4, 8, 16, 32, 64, 128, 256, 512, 1024]
+        ref_channels = [512, 512, 512, 512, 512, 256, 128, 64, 32, 16]
+        generator = Generator()
+
+        for res, ref in zip(resolutions, ref_channels):
+            self.assertEqual(generator._get_out_channels(res), ref)
+
+    def test_forward_without_fade_in(self):
+        with torch.no_grad():
+            generator = Generator()
+            resolutions = [4, 8, 16, 32, 64, 128, 256, 512, 1024]
+
+            B, C, H, W = 1, 512, 1, 1
+            x = torch.randn(B, C, H, W)
+
+            for res in resolutions:
+                generator.set_output_image_size(res)
+                result = generator(x)
+                self.assertEqual(
+                    result.shape, (B, 3, res, res))
+
+    def test_forward_with_fade_in(self):
+        with torch.no_grad():
+            generator = Generator()
+            generator.fade_in_weight = 0.5
+            resolutions = [4, 8, 16, 32, 64, 128, 256, 512, 1024]
+
+            B, C, H, W = 3, 512, 1, 1
+            x = torch.randn(B, C, H, W)
+
+            for res in resolutions:
+                generator.set_output_image_size(res)
+                result = generator(x)
+                self.assertEqual(
+                    result.shape, (B, 3, res, res))
+
+    def test_pixel_wise_norm_method(self):
```

Comment:
의미를 명확히 하기 위해 normalization이라고 표기하는 것이 좋겠습니다.

```suggestion
    def test_pixelwise_normalization_method(self):
```